internal/api/grpc/server/middleware/execution_interceptor.go

```diff
@@ -89,18 +89,27 @@ type ContextInfoRequest struct {
 }
 
 func (c *ContextInfoRequest) GetHTTPRequestBody() []byte {
-	data, err := json.Marshal(c)
+	jsonpb := &runtime.JSONPb{
+		UnmarshalOptions: protojson.UnmarshalOptions{
+			DiscardUnknown: true,
+			AllowPartial:   true,
+		},
+	}
```

Comment:
Please initialize a single (un)marshaller that can be reused in all functions.
Also I would directly use protopb instead of the gateway wrapper.